web/pandas/pdeps/0013-logical-type-system.md

```diff
@@ -0,0 +1,196 @@
+
+  - pd.DecimalDtype()
+  - pd.DateDtype()
+  - pd.DatetimeDtype(unit, tz)
+  - pd.Duration()
+  - pd.CalendarInterval()
+  - pd.BinaryDtype()
+  - pd.DictDtype()
+  - pd.ListDtype()
+  - pd.StructDtype()
+  - pd.ObjectDtype()
+  - pd.NullDtype()
+
+The storage / backend to each of these types is left as an implementation detail. The fact that ``pd.StringDtype()`` may be backed by Arrow while ``pd.PeriodDtype()`` continues to be a custom solution is of no concern to the end user. Over time this will allow us to adopt more Arrow behind the scenes without breaking the front end for our end users, but _still_ giving us the flexibility to produce data types that Arrow will not implement (e.g. ``pd.ObjectDtype()``).
+
+The methods of each logical type are expected in turn to yield another logical type. This can enable us to smooth over differences between the NumPy and Arrow world, while also leveraging the best of both backends. To illustrate, let's look at some methods where the return type today deviates for end users depending on if they are using NumPy-backed data types or Arrow-backed data types. The equivalent PDEP-13 logical data type is presented as the last column:
+
+| Method             | NumPy-backed result | Arrow Backed result type | PDEP-13 result type            |
+|--------------------|---------------------|--------------------------|--------------------------------|
+| Series.str.len()   | np.float64          | pa.int64()               | pd.Int64Dtype()                |
+| Series.str.split() | object              | pa.list(pa.string())     | pd.ListDtype(pa.StringDtype()) |
+| Series.dt.date     | object              | pa.date32()              | pd.DateDtype()                 |
+
+The ``Series.dt.date`` example is worth an extra look - with a PDEP-13 logical type system in place we would theoretically have the ability to keep our default ``pd.DatetimeDtype()`` backed by our current NumPy-based array but leverage pyarrow for the ``Series.dt.date`` solution, rather than having to implement a DateArray ourselves.
+
+To implement this PDEP, we expect all of the logical types to have at least the following metadata:
+
+    * storage: Either "numpy" or "pyarrow". Describes the library used to create the data buffer
+    * physical_type: Can expose the physical type being used. As an example, StringDtype could return pa.string_view
+    * na_value: Either pd.NA, np.nan, or pd.NaT.
+
+While these attributes are exposed as construction arguments to end users, users are highly discouraged from trying to control them directly. Put explicitly, this PDEP allows a user to request a ``pd.XXXDtype(storage="numpy")`` to request a NumPy-backed array, if possible. While pandas may respect that during construction, operations against that data make no guarantees that the storage backend will be persisted through, giving pandas the freedom to convert to whichever storage is internally optimal (Arrow will typically be preferred).
```

Comment:
Should each dtype necessarily expose each of those metadata items as a construction argument? If there is only one valid option (eg a dtype that is implemented only using numpy or pyarrow for storage, or that has only a single physical type), it seems not necessary to expose this as argument?